packages/build-utils/src/executors/old-builder/old-executor.ts

```diff
@@ -1,10 +1,23 @@
-import { ExecutorContext } from '@nx/devkit';
+import { addDependenciesToPackageJson, ExecutorContext, Tree, workspaceRoot } from '@nx/devkit';
 import { z } from 'zod';
-import { stat, mkdir, copyFile, unlink } from 'fs';
+import { chmodSync, copyFile, existsSync, lstatSync, mkdir, readFileSync, readdirSync, renameSync, stat, unlink, writeFileSync } from 'fs';
 import { copy } from 'fs-extra';
 import { promisify } from 'util';
 import { execSync } from 'child_process';
 
+// Typescript prevents us from importing the global package.json to check which dependencies we want shared
+// See https://github.com/microsoft/TypeScript/issues/9858
+const rawData = readFileSync(`${workspaceRoot}/package.json`, {
+  encoding: 'utf8',
+});
+const globalPackageJson = JSON.parse(rawData);
+const globalDependencies = globalPackageJson.dependencies;
+
+// Filters the globally hoisted dependencies to be added to the packages published package.json (outside NX scope)
+// TODO add tslib
+const sharedGlobalDependencies = (({ axios }) => ({ axios }))(globalDependencies);
+const sharedGlobalDevDependencies = {};
```

Comment:
FWIW in the future we may be able to utilize something like https://nx.dev/nx-api/esbuild/executors/esbuild#generatepackagejson and/or https://nx.dev/concepts/buildable-and-publishable-libraries#publishable-libraries